evidence/pool.go

```diff
@@ -48,11 +48,10 @@ type Pool struct {
 
 // NewPool creates an evidence pool. If using an existing evidence store,
 // it will add all pending evidence to the concurrent list.
-func NewPool(evidenceDB dbm.DB, stateDB sm.Store, blockStore BlockStore) (*Pool, error) {
-
+func NewPool(logger log.Logger, evidenceDB dbm.DB, stateDB sm.Store, blockStore BlockStore) (*Pool, error) {
```

Comment:
Are we actually adding this logger to the evidence pool though? Further down in the constructor we have
```go
...
logger:   log.NewNopLogger(),
...
```

Reply:
Also just out of curiosity, I would normally append a new argument to the end (or try order them from most important to least important). Is there a reason why you put it at the start?

Reply:
> Are we actually adding this logger to the evidence pool though? Further down in the constructor we have

🤦  good catch. I didn't see this from the original code. Thank you.

Reply:
From my experience, objects like loggers, contexts and configurations come first.

But in general, arguments should be grouped by types and then by pointer semantics. But there isn't a hard rule to this AFAIK.